scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -2130,39 +2157,44 @@ private void sendMessage(final long nanos, final RawData message, final Connecti
 						return;
 					}
 					message.onConnecting();
-					Handshaker handshaker;
 					SessionId sessionId;
 					if (session != null) {
 						sessionId = session.getSessionIdentifier();
 						ticket = session.getSessionTicket();
-						connectionStore.removeFromEstablishedSessions(session, connection);
+						if (!probing) {
+							connectionStore.removeFromEstablishedSessions(session, connection);
+						}
 					} else {
 						// then ticket != null
 						sessionId = connection.getSessionIdentity();
 					}
-					connection.resetSession();
-					Handshaker previous = connection.getOngoingHandshake();
+					if (probing) {
+						connection.setResumptionRequired(false);
```

Comment:
This is the end of the week so maybe I'm a bit tired, but the code seems hard to read several line above we just `connection.setResumptionRequired(true)` if probing is true and now we set it to false because probing is true

And until now I believed that probing was only for abbreviated handshake (so resumption = true) :thinking:

Reply:
Above, the `connection.setResumptionRequired(true)` was introduced to trigger the later
`if (connection.isAutoResumptionRequired(timeout))`. That was implemented to support message specific handshake behaviour.

```
if (probing) {
	connection.setResumptionRequired(false);
} else {
	connection.resetSession();
}
```

Here the `setResumptionRequired(false)` in case of probing replaces the `resetSession()`, which also includes that. The difference is to keep the session.